dek-registry-client/src/main/java/io/confluent/dekregistry/client/CachedDekRegistryClient.java

```diff
@@ -0,0 +1,293 @@
+      }
+      throw new RuntimeException(e.getCause());
+    }
+  }
+
+  @Override
+  public List<String> listDeks(String kekName, boolean lookupDeleted)
+      throws IOException, RestClientException {
+    return restService.listDeks(kekName, lookupDeleted);
+  }
+
+  @Override
+  public Dek getDek(String name, String subject, boolean lookupDeleted)
+      throws IOException, RestClientException {
+    return getDek(name, subject, null, lookupDeleted);
+  }
+
+  @Override
+  public Dek getDek(String name, String subject, DekFormat algorithm, boolean lookupDeleted)
+      throws IOException, RestClientException {
+    try {
+      return dekCache.get(new DekId(name, subject, algorithm), () ->
+          restService.getDek(name, subject, algorithm, lookupDeleted));
+    } catch (ExecutionException e) {
+      if (e.getCause() instanceof IOException) {
+        throw (IOException) e.getCause();
+      } else if (e.getCause() instanceof RestClientException) {
+        throw (RestClientException) e.getCause();
+      }
+      throw new RuntimeException(e.getCause());
+    }
+  }
+
+  @Override
+  public Kek createKek(
+      String name,
+      String kmsType,
+      String kmsKeyId,
+      Map<String, String> kmsProps,
+      String doc,
+      boolean shared)
+      throws IOException, RestClientException {
+    CreateKekRequest request = new CreateKekRequest();
+    request.setName(name);
+    request.setKmsType(kmsType);
+    request.setKmsKeyid(kmsKeyId);
+    request.setKmsProps(kmsProps);
+    request.setDoc(doc);
+    request.setShared(shared);
+    Kek kek = restService.createKek(request);
+    kekCache.put(new KekId(name), kek);
+    return kek;
+  }
+
+  @Override
+  public Dek createDek(
+      String kekName,
+      String subject,
+      DekFormat algorithm,
+      String encryptedKeyMaterial)
+      throws IOException, RestClientException {
+    CreateDekRequest request = new CreateDekRequest();
+    request.setSubject(subject);
+    request.setAlgorithm(algorithm);
+    request.setEncryptedKeyMaterial(encryptedKeyMaterial);
+    Dek dek = restService.createDek(kekName, request);
+    dekCache.put(new DekId(kekName, subject, algorithm), dek);
+    return dek;
+  }
+
+  @Override
+  public Kek updateKek(
+      String name,
+      Map<String, String> kmsProps,
+      String doc,
+      Boolean shared)
+      throws IOException, RestClientException {
+    UpdateKekRequest request = new UpdateKekRequest();
+    request.setKmsProps(kmsProps);
+    request.setDoc(doc);
+    request.setShared(shared);
+    Kek kek = restService.updateKek(name, request);
+    kekCache.put(new KekId(name), kek);
+    return kek;
+  }
+
+  @Override
+  public void deleteKek(String name, boolean permanentDelete)
+      throws IOException, RestClientException {
+    restService.deleteKek(name, permanentDelete);
+    kekCache.invalidate(new KekId(name));
+  }
+
+  @Override
+  public void deleteDek(String name, String subject, boolean permanentDelete)
```

Comment:
can we rename this to `kekName` here and everywhere else?

Reply:
Good idea